### ISSUE TITLE
[BUG] Azure.ResourceManager.Quota CreateOrUpdate fails when WaitUntil.Completed is set

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/azure-sdk-for-net#45573]
### Library name and version

Azure.ResourceManager.Quota latest

### Describe the bug

The underling [LRO logic](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs#L389-L410) assumes that for LROs initiated with PATCH/PUT that you GET the originating URL for the final response, ignoring the `final-state-via` setting. As a result, an exception is thrown when waiting for this LRO to complete.

### Expected behavior

LRO completes.

### Actual behavior

An exception is thrown when attempting to wait for the LRO to complete.

### Reproduction Steps

Execute test [SetGroupQuota](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/quota/Azure.ResourceManager.Quota/tests/Tests/GroupQuotaTests.cs#L32) in live mode and change line 59 to the following.
```c#
var createResponse  = (await collection.CreateOrUpdateAsync(WaitUntil.Completed, groupQuotaName, data)).Value;
```

**Comment by jsquire** [comment]
//cc: @annelo-msft

**Comment by annelo-msft** [comment]
Per offline discussion, `Final-State-Via` was designed for use with POST, and not intended to be used outside this context.  It would be best if the service could standard ARM guidance and either use POST or remove the Location header - if so, I would expect the C# component to function as intended.

**Comment by jhendrixMSFT** [comment]
Per discussion, `final-state-via` is for `POST` only. The docs have been updated, see https://github.com/Azure/autorest/pull/5004.

**Comment by annelo-msft** [comment]
Thanks for updating those docs, @jhendrixMSFT!

I don't have a good sense of where the guidance for ARM LROs live holistically, but I was also looking at [these docs here](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#asynchronous-api-reference) to see if I could understand the intended flow better.  Would it be worth adding something to these docs as well?  Do you know what is the intended source of truth for ARM LRO guidance?

**Comment by JeffreyRichter** [comment]
Yes, what Joel says is correct. For data-plane the LRO guidelines Joel linked to is the most current.
ARM has been slow to adopt and tweak to their needs.
